x-pack/plugins/upgrade_assistant/public/components/tabs/checkup/deprecations/index_table.tsx

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { sortBy } from 'lodash';
+import React from 'react';
+
+import { EuiBasicTable, EuiButton } from '@elastic/eui';
+import { injectI18n } from '@kbn/i18n/react';
+
+const PAGE_SIZES = [10, 25, 50, 100, 250, 500, 1000];
+
+export interface IndexDeprecationDetails {
+  index: string;
+  details?: string;
+  actions?: Array<{
+    label: string;
+    url: string;
+  }>;
+}
+
+export interface IndexDeprecationTableProps extends ReactIntl.InjectedIntlProps {
+  indices: IndexDeprecationDetails[];
+}
+
+interface IndexDeprecationTableState {
+  sortField: string;
+  sortDirection: 'asc' | 'desc';
+  pageIndex: number;
+  pageSize: number;
+}
+
+export class IndexDeprecationTableUI extends React.Component<
+  IndexDeprecationTableProps,
+  IndexDeprecationTableState
+> {
+  constructor(props: IndexDeprecationTableProps) {
+    super(props);
+
+    this.state = {
+      sortField: 'index',
+      sortDirection: 'asc',
+      pageIndex: 0,
+      pageSize: 10,
+    };
+  }
+
+  public render() {
+    const { intl } = this.props;
+    const { pageIndex, pageSize, sortField, sortDirection } = this.state;
+
+    const columns = [
+      {
+        field: 'index',
+        name: intl.formatMessage({
+          id: 'xpack.upgradeAssistant.checkupTab.deprecations.indexTable.indexColumnLabel',
+          defaultMessage: 'Index',
+        }),
+        sortable: true,
+      },
+      {
+        field: 'details',
+        name: intl.formatMessage({
+          id: 'xpack.upgradeAssistant.checkupTab.deprecations.indexTable.detailsColumnLabel',
+          defaultMessage: 'Details',
+        }),
+      },
+    ];
+
+    if (this.actionsColumn) {
+      // @ts-ignore
+      columns.push(this.actionsColumn);
+    }
+
+    const sorting = { sort: { field: sortField, direction: sortDirection } };
+    const pagination = {
+      pageIndex,
+      pageSize,
+      ...this.pageSizeOptions(),
+    };
+
+    return (
+      <EuiBasicTable
+        items={this.getRows()}
+        columns={columns}
+        sorting={sorting}
+        pagination={pagination}
+        onChange={this.onTableChange}
+      />
```

Comment:
Add `hasActions={false}` so that the buttons aren't desaturated. They currently look disabled.

 
<img width="983" alt="screen shot 2018-12-11 at 09 57 59 am" src="https://user-images.githubusercontent.com/549577/49810877-1e18cf00-fd30-11e8-820b-925e37edb281.png">